cmd/analyzer/analyzer.go

```diff
@@ -238,21 +240,73 @@ func NewService(cfg *config.AnalysisConfig) (*Service, error) {
 		return nil, err
 	}
 
-	// Initialize analyzers.
-	analyzers := map[string]A{}
+	// Initialize fast-sync analyzers.
+	fastSyncAnalyzers := []A{}
 	if cfg.Analyzers.Consensus != nil {
-		analyzers, err = addAnalyzer(analyzers, err, func() (A, error) {
-			startHeight := int64(cfg.Analyzers.Consensus.From)
-			startRecord, err1 := cfg.Source.History().RecordForHeight(startHeight)
-			if err1 != nil {
-				return nil, fmt.Errorf("getting history record for consensus starting block %d: %w", startHeight, err1)
+		if fastRange := cfg.Analyzers.Consensus.FastSyncRange(); fastRange != nil {
+			for i := 0; i < cfg.Analyzers.Consensus.FastSync.Parallelism; i++ {
+				fastSyncAnalyzers, err = addAnalyzer(fastSyncAnalyzers, err, func() (A, error) {
+					sourceClient, err1 := sources.Consensus(ctx)
+					if err1 != nil {
+						return nil, err1
+					}
+					return consensus.NewAnalyzer(*fastRange, cfg.Analyzers.Consensus.BatchSize, analyzer.FastSyncMode, *cfg.Source.History(), sourceClient, dbClient, logger)
+				})
+			}
+		}
+	}
+	if cfg.Analyzers.Emerald != nil {
+		if fastRange := cfg.Analyzers.Emerald.FastSyncRange(); fastRange != nil {
+			for i := 0; i < cfg.Analyzers.Emerald.FastSync.Parallelism; i++ {
```

Comment:
Could this repeated initialization be moved into a helper method, that gets the `BlockBasedAnalyzerConfig` and `Runtime` passed in?